test/ha/ha.go

```diff
@@ -26,34 +26,49 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/sets"
 	"k8s.io/apimachinery/pkg/util/wait"
+
 	"knative.dev/pkg/test"
 	"knative.dev/pkg/test/logging"
 )
 
+func countingRFind(wr rune, wc int) func(rune) bool {
+	cnt := 0
+	return func(r rune) bool {
+		if r == wr {
+			cnt++
+		}
+		return cnt == wc
+	}
+}
+
+func extractDeployment(pod string) string {
+	if x := strings.LastIndexFunc(pod, countingRFind('-', 2)); x != -1 {
+		return pod[:x]
+	}
+	return ""
+}
+
 // GetLeaders collects all of the leader pods from the specified deployment.
+// GetLeaders will return duplicate pods by design.
 func GetLeaders(t *testing.T, client *test.KubeClient, deploymentName, namespace string) ([]string, error) {
 	leases, err := client.Kube.CoordinationV1().Leases(namespace).List(metav1.ListOptions{})
 	if err != nil {
 		return nil, fmt.Errorf("error getting leases for deployment %q: %w", deploymentName, err)
 	}
-	var pods []string
+	ret := make([]string, 0, len(leases.Items))
```

Comment:
Which is the greater evil: overallocation or resizing? 🤔

Reply:
resizing, since resize will probably overallocate as well! (newSize *=2 most of the time).